src/lib.rs

```diff
@@ -885,51 +925,747 @@ impl GooseAttack {
+                        "The --no-hash-check flag can not be set together with the --worker flag."
+                            .to_string(),
+                });
+            }
+        }
+
+        // Otherwise run in standalone attack mode.
+        if self.attack_mode == GooseMode::Undefined {
+            self.attack_mode = GooseMode::StandAlone;
+
+            if self.configuration.no_hash_check {
+                return Err(GooseError::InvalidOption {
+                    option: "--no-hash-check".to_string(),
+                    value: self.configuration.no_hash_check.to_string(),
+                    detail: "The --no-hash-check flag can not be set without also setting the --manager flag.".to_string(),
+                });
+            }
+        }
+
+        Ok(())
+    }
+
+    // Determine how many workers to expect.
+    fn set_expect_workers(&mut self) -> Result<(), GooseError> {
+        let expect_workers = if self.configuration.expect_workers > 0 {
+            self.configuration.expect_workers
+        } else if let Some(number) = self.defaults.expect_workers {
+            // Only set expect_workers from default if on Manager.
+            if self.attack_mode == GooseMode::Manager {
+                number
+            } else {
+                0
+            }
+        } else {
+            0
+        };
+
+        // Generally disallow --expect-workers without --master.
+        if self.attack_mode != GooseMode::Manager && expect_workers > 0 {
+            return Err(GooseError::InvalidOption {
+                option: "--expect-workers".to_string(),
+                value: self.configuration.expect_workers.to_string(),
+                detail: "The --expect-workers flag can not be set without also setting the --manager flag.".to_string(),
+            });
+        }
+
+        if self.attack_mode == GooseMode::Manager {
+            // Must expect at least 1 Worker when running as Manager.
+            if expect_workers < 1 {
+                return Err(GooseError::InvalidOption {
+                    option: "--expect-workers".to_string(),
+                    value: expect_workers.to_string(),
+                    detail: "The --expect-workers option must be set to at least 1.".to_string(),
+                });
+            }
+
+            // Must not expect more Workers than Users.
+            if expect_workers as usize > self.users {
+                return Err(GooseError::InvalidOption {
+                    option: "--expect-workers".to_string(),
+                    value: expect_workers.to_string(),
+                    detail: "The --expect-workers option can not be set to a value larger than --users option.".to_string(),
+                });
+            }
+        }
+
+        // Overload configuration.expect_workers to make available in Worker process.
+        self.configuration.expect_workers = expect_workers;
+
+        Ok(())
+    }
+
+    fn set_gaggle_host_and_port(&mut self) -> Result<(), GooseError> {
```

Comment:
@LionsAd Again we overload the configuration structure (this simplifies passing the information around in a Gaggle)